src/Components/ModuleForm/ModuleForm.js

```diff
@@ -0,0 +1,310 @@
+import React, { Component } from "react";
+import "./ModuleForm.css";
+import { Button, ListGroup, Card } from "react-bootstrap";
+
+class ModuleForm extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      enabled: "1",
+      userAction: "Add",
+      moduleName: "",
+      moduleList: JSON.parse(localStorage.getItem("moduleList")) || [],
+      grade: "A",
+      error: "",
+    };
+  }
+
+  handleSubmit = (event) => {
+    const { userAction, moduleList, moduleName, grade } = this.state;
+
+    console.log("handleSubmit");
+
+    if (moduleName === "") {
+      this.setState({
+        error: "Module Name cannot be empty",
+      });
+      event.preventDefault();
+      return;
+    }
+    // Add logic for adding modules
+    if (userAction === "Add") {
+      const newModule = { moduleName, grade };
+      const searchTerm = moduleName;
+      const existingModuleList = moduleList.filter(
```

Comment:
I would assume that this is only filtering a single item of the list. In that case using `.find()` might be more appropriate